pkg/controller/autoscaling/elasticsearch/validation/webhook_test.go

```diff
@@ -0,0 +1,606 @@
+									CPURange:       &commonv1alpha1.QuantityRange{Min: resource.MustParse("1"), Max: resource.MustParse("2")},
+									MemoryRange:    &commonv1alpha1.QuantityRange{Min: resource.MustParse("2Gi"), Max: resource.MustParse("2Gi")},
+									StorageRange:   &commonv1alpha1.QuantityRange{Min: resource.MustParse("5Gi"), Max: resource.MustParse("10Gi")},
+									NodeCountRange: commonv1alpha1.CountRange{Min: 5, Max: 4},
+								},
+							},
+						},
+					},
+				},
+				checker: yesCheck,
+			},
+			wantValidationError: ptr.String("spec.policies[0].resources.nodeCount.max: Invalid value: 4: max node count must be an integer greater or equal than the min node count"),
+		},
+		{
+			name: "Min. CPU is greater than max",
+			args: args{
+				es: es(map[string]string{}, map[string][]string{"nodeset-data-1": {"data"}}, nil, "8.0.0"),
+				esa: v1alpha1.ElasticsearchAutoscaler{
+					ObjectMeta: metav1.ObjectMeta{Name: "esa", Namespace: "ns"},
+					Spec: v1alpha1.ElasticsearchAutoscalerSpec{
+						ElasticsearchRef: v1alpha1.ElasticsearchRef{
+							Name: "es",
+						},
+						AutoscalingPolicySpecs: commonv1alpha1.AutoscalingPolicySpecs{
+							{
+								NamedAutoscalingPolicy: commonv1alpha1.NamedAutoscalingPolicy{
+									Name:              "data_policy",
+									AutoscalingPolicy: commonv1alpha1.AutoscalingPolicy{Roles: []string{"data"}},
+								},
+								AutoscalingResources: commonv1alpha1.AutoscalingResources{
+									CPURange:       &commonv1alpha1.QuantityRange{Min: resource.MustParse("3"), Max: resource.MustParse("2")},
+									MemoryRange:    &commonv1alpha1.QuantityRange{Min: resource.MustParse("2Gi"), Max: resource.MustParse("2Gi")},
+									StorageRange:   &commonv1alpha1.QuantityRange{Min: resource.MustParse("5Gi"), Max: resource.MustParse("10Gi")},
+									NodeCountRange: commonv1alpha1.CountRange{Min: 2, Max: 4},
+								},
+							},
+						},
+					},
+				},
+				checker: yesCheck,
+			},
+			wantValidationError: ptr.String("spec.policies[0].cpu.max: Invalid value: \"2\": max quantity must be greater or equal than min quantity"),
+		},
+		// Volumes validations
+		{
+			name: "Not the default volume claim",
+			args: args{
+				es: es(map[string]string{}, map[string][]string{"nodeset-data-1": {"data"}}, []string{"volume1"}, "8.0.0"),
+				esa: v1alpha1.ElasticsearchAutoscaler{
+					ObjectMeta: metav1.ObjectMeta{Name: "esa", Namespace: "ns"},
+					Spec: v1alpha1.ElasticsearchAutoscalerSpec{
+						ElasticsearchRef: v1alpha1.ElasticsearchRef{
+							Name: "es",
+						},
+						AutoscalingPolicySpecs: commonv1alpha1.AutoscalingPolicySpecs{
+							{
+								NamedAutoscalingPolicy: commonv1alpha1.NamedAutoscalingPolicy{
+									Name:              "data_policy",
+									AutoscalingPolicy: commonv1alpha1.AutoscalingPolicy{Roles: []string{"data"}},
+								},
+								AutoscalingResources: defaultResources,
+							},
+						},
+					},
+				},
+				checker: yesCheck,
+			},
```

Comment:
Missing the `wantValidationError` here?

Reply:
We actually support the case where the user is not relying on the default volume claim. I added a comment in the unit test [here](https://github.com/elastic/cloud-on-k8s/pull/5978/commits/285485c5e70d3eece4e1544a5f0b7b21a49c3a24#diff-f8e73750d745a6d002f0e88a8c1cd42455668bc2374110c3a77105e43ef50da9R483).